server/src/main/java/org/elasticsearch/persistent/PersistentTasksService.java

```diff
@@ -208,11 +209,14 @@ ThreadPool getThreadPool() {
     /**
      * Waits for persistent tasks to comply with a given predicate, then call back the listener accordingly.
      *
-     * @param predicate the predicate to evaluate
+     * @param projectId the project that the persistent tasks are associated with
+     * @param predicate the predicate to evaluate, must be able to handle {@code null} input which means either the project
+     *                  does not exist or persistent tasks for the project do not exist
      * @param timeout a timeout for waiting
      * @param listener the callback listener
      */
     public void waitForPersistentTasksCondition(
```

Comment:
Is there a reason you didn't also update the (overloaded) `waitForPersistentTasksCondition` method above? I think it makes sense to do both within this PR.

Reply:
The other method is not an overload, it is `waitForPersistentTaskCondition`, note the singular `Task`. I didn't touch that because you handled it in #124000?